contractcourt/channel_arbitrator.go

```diff
@@ -2378,7 +2446,14 @@ func (c *ChannelArbitrator) prepContractResolutions(
 
 		// If we can fail an HTLC immediately (an outgoing HTLC with no
 		// contract), then we'll assemble an HTLC fail packet to send.
-		case HtlcFailNowAction:
+		//
+		// NOTE: In case we initiated the force close locally all the
+		// outgoing dust htlcs are already failed backwards when the
+		// commitment transaction is confirmed. However if the remote
+		// confirms or we broadcast the local commitment by some other
+		// means we need to fail them back here as well to be sure all
+		// outgoing dust htlcs resolve their incoming htlc.
+		case HtlcFailDanglingAction, HtlcFailDustAction:
```

Comment:
Maybe we can remove this logic from `prepContractResolutions` - it's not related to contract resolutions anyway. Then we can call the new `cancelDustPreMaturely` wherever needed.

Reply:
thats a very good point.

Reply:
persisting the CancledHTLCs right now in the ArbLog, because I did not see another way to make sure we do not query the CircuitMap more than once and hence cancel back htlc twice which leads to log errors.